ARIA alerts: clarify meaning of static vs dynamic (#13458)
mdn/content: files/en-us/web/accessibility/aria/roles/alert_role/index.md

- The user's login session is about to expire
- The connection to the server was lost so local changes will not be saved

> The `alert` role should only be used for static text content, not interactive elements such as links or buttons. The element with the `alert` role does not have to be able to receive focus, as screen readers (speech or braille) will automatically announce the updated content regardless of where keyboard focus when the role is added.

Reviewer: I think the word "static" is part of the confusion, and the addition of the "not interactive..." aspect you added I think helps better clarify what shouldn't go in the element

```suggestion
The `alert` role should only be used for text content, not interactive elements such as links or buttons. The element with the `alert` role does not have to be able to receive focus, as screen readers (speech or braille) will automatically announce the updated content regardless of where keyboard focus when the role is added.
```

> **Warning:** Because of its intrusive nature, the `alert` role must be used sparingly and only in situations where the user's immediate attention is required.

The [`alert`](https://www.w3.org/TR/wai-aria-1.1/#alert) role is of the five [live region](/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) roles. Dynamic changes that are less urgent should use a less aggressive method, such as including `aria-live="polite"` or using an other live region role like [`status`](/en-US/docs/Web/Accessibility/ARIA/Roles/status_role). If the user is expected to close the alert, then the [`alertdialog`](/en-US/docs/Web/Accessibility/ARIA/Roles/alertdialog_role) role should be used instead.

> The most important thing to know about the `alert` role is that it's for dynamic content that is displayed after the page has loaded. It is perfect for situations such as when a user fills out a form and JavaScript is used to add an error message - the alert would immediately read out the message. It should not be used on HTML that the user hasn't interacted with. For example, if a page loads with multiple visible alerts scattered throughout, no alert would be read as they were not dynamically triggered.

Reviewer: The first sentence might still be a little confusing on what "dynamic content" is. Suggested an alternative phrasing.

I also think the last sentence example is a little confusing, so I made a suggestion on the distinction between the concept of an alert and the usage of an alert role.

```suggestion
The most important thing to know about the `alert` role is that it's for content that is dynamically displayed, not for content that appears on page load. It is perfect for situations such as when a user fills out a form and JavaScript is used to add an error message - the alert would immediately read out the message. It should not be used on HTML that the user hasn't interacted with. For example, if a page loads with multiple visible alerts scattered throughout, the alert role should not be used, as the messages were not dynamically triggered.
```

Author: Thanks for your suggestions, I've approved them 🙂
